### PR TITLE
[base] Optimize preview snapshot fetching

### PR DESCRIPTION
**Pull request by rexxars**
When resolving previews, we often end up with queries such as:
```js
[
  *[_id==$id_0]{_id,_type,url},
  *[_id==$id_1]{_id,_type,url},
  *[_id==$id_2]{_id,_type,url},
  *[_id==$id_3]{_id,_type,url},
  *[_id==$id_4]{_id,_type,url},
  *[_id==$id_5]{_id,_type,url}
]
```

Which currently is not optimized on the server side. This PR optimizes the cases where the projects are equal, such as in the case above, and instead creates a query like the following:

```js
*[_id in ["someId_0", "someId_1", "someId_2", "someId_3", "someId_4"]]{_id, _type, url}
```

When the response arrives, we reassemble the array so that the results are in the same order as the selection passed to the `observeWithPaths`.

I'm sure this could be solved slightly cleaner, but this reduces a 15-document query from ~250ms to ~15ms.

### CODE REVIEW

**Comment by bjoerge** [comment]
Looks good!
Seems like it only enter the optimized code path if _all_ selections have identical paths? The `fetchAllDocumentSnapshots()`-function will often be called with a mix of path selections, so an improvement could be to group the selections with identical paths within the same query.

E.g. so the given selections:
```
[
  ['doc1', ['fieldA', 'fieldB', 'fieldC.fieldCA']],
  ['doc2', ['fieldA', 'fieldB', 'fieldC.fieldCA']],
  ['doc7', ['fieldX', 'fieldY']],
  ['doc3', ['fieldA', 'fieldB', 'fieldC.fieldCA']]
]
```
would yield the query
```
[
  *[_id in ["doc1","doc2","doc3"]]{_id,_type,fieldA,fieldB,fieldC.fieldCA},
  *[_id in ["doc7"]]{_id,_type,fieldX,fieldY}
]
```

**Comment by bjoerge** [comment]
Couldn't get it out of my head, so I rewrote it a bit, hope you don't mind. It now optimizes all the unique projections within the same query.

**Comment by bjoerge** [comment]
Oh, just realized that the `_id in [...]` doesn't return hits in the order specified by the list of ids. So there's a bug in her now. Fixing.

File: packages/@sanity/base/src/preview/observeWithPaths.js [hunk 1]
```diff
@@ -18,17 +19,51 @@ function listen(id) {
...
 }
 
-function fetchAllDocumentSnapshots(selections) {
-  const optimizedParams = {}
-  const queryParts = selections.map(([id, paths], queryIndex) => {
-    optimizedParams[[`id_${queryIndex}`]] = id
-    return `*[_id==$id_${queryIndex}]{_id,_type,${paths.join(',')}}`
-  })
+function combineSelections(selections) {
+  return Object.values(selections.reduce((output, [id, paths], index) => {
```
Comment (rexxars) [inline packages/@sanity/base/src/preview/observeWithPaths.js]: `Object.values()` is not supported in IE11, perhaps use `values` from lodash?

Comment (bjoerge) [reply packages/@sanity/base/src/preview/observeWithPaths.js]: Good idea. Will fix!

**Comment by rexxars** [comment]
I can't approve "my own" PR, so I'll just say LGTM!

**Pull request merged**